Awful.apk/src/main/java/com/ferg/awfulapp/preferences/SettingsActivity.java

```diff
@@ -302,32 +302,60 @@ protected void importFile(Intent data) {
 
     @Override
     protected Dialog onCreateDialog(int dialogId) {
-        switch (dialogId) {
-            case DIALOG_ABOUT:
-                CharSequence app_version = getText(R.string.app_name);
-                try {
-                    app_version = app_version + " " +
-                            getPackageManager().getPackageInfo(getPackageName(), 0)
-                                    .versionName;
-                } catch (PackageManager.NameNotFoundException e) {
-                    // rather unlikely, just show app_name without version
-                }
-                // Build the text for the About dialog
-                Resources res = getResources();
-                String aboutText = getString(R.string.about_contributors_title) + "\n\n";
-                aboutText += StringUtils.join(res.getStringArray(R.array.about_contributors_array), '\n');
-                aboutText += "\n\n" + getString(R.string.about_libraries_title) + "\n\n";
-                aboutText += StringUtils.join(res.getStringArray(R.array.about_libraries_array), '\n');
-
-                return new AlertDialog.Builder(this)
-                        .setTitle(app_version)
-                        .setMessage(aboutText)
-                        .setNeutralButton(android.R.string.ok, (dialog, which) -> {
-                        })
-                        .create();
-            default:
-                return super.onCreateDialog(dialogId);
+        if (dialogId == DIALOG_ABOUT)
```

Comment:
Making the ``switch`` an ``if/else`` is cool. I'm honestly not a fan of the ``StringBuilder`` stuff though, it's really verbose and a lot less clear what's happening - I don't think the efficiency gain is worth it here

If you keep the original message building and dialog creation code, but stick that in ``#getAboutDialog`` as a single method I think that's probably neatest and makes it cleaner to add stuff later. Also can you add a heading that says "User Agent" or something before that section, just so people know what it is?